src/taskEvents/utils.js

```diff
@@ -107,6 +111,99 @@ async function getData(data) {
   };
 }
 
+async function getUserSkills(id) {
+  try {
+    const res = await fetch(`${BASE_URL}/users/${id}/skills`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+
+    const userSkills = await res.json();
+    return userSkills;
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function getTagLevelOptions() {
+  try {
+    const levelsResponse = await fetch(`${BASE_URL}/levels`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+    const tagsResponse = await fetch(`${BASE_URL}/tags`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+
+    const { allTags } = await tagsResponse.json();
+    let { allLevels } = await levelsResponse.json();
+    allLevels = allLevels.sort((a, b) => {
+      if (parseInt(a.name) < parseInt(b.name)) return -1;
+      if (parseInt(a.name) > parseInt(b.name)) return 1;
+      return 0;
+    });
+    return { allLevels, allTags };
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function addSkillToUser(skillToAdd, levelToAdd, userid) {
+  const body = {
+    itemid: userid,
+    itemType: 'USER',
+    tagPayload: [
+      {
+        levelid: levelToAdd?.id,
+        tagid: skillToAdd?.id,
+      },
+    ],
+  };
+  try {
+    const response = await fetch(`${BASE_URL}/items`, {
+      method: 'POST',
+      credentials: 'include',
+      body: JSON.stringify(body),
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+    return response;
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function removeSkillFromUser(tagid, userid) {
+  const body = {
+    itemid: userid,
+    tagid,
+  };
+  try {
+    const response = await fetch(`${BASE_URL}/items`, {
+      method: 'DELETE',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+      body: JSON.stringify(body),
+    });
+    return response;
+  } catch (error) {
+    console.log(error);
```

Comment:
can we put alert here for now? if the request fails then the user knows that something has went wrong

Reply:
yes added alert